### PR TITLE
Dependency managers support

### LINKED ISSUE #3: Add Carthage support
Currently we support only SPM, we should also support Carthage.

### PR DESCRIPTION
**Pull request by LukasHromadnik**
In this PR we added support for 3rd dependency managers:
* Cocoapods
* Carthage

The whole release is possible to be done using Tapestry.

Some code had to be updated to be buildable for macOS platform.

This PR resolves #3 and #4

### CODE REVIEW

File: OpenGoogleSignInSDK.podspec [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+Pod::Spec.new do |s|
+  s.name = "OpenGoogleSignInSDK"
+  s.version = "1.0.0"
```
Comment (olejnjak) [inline OpenGoogleSignInSDK.podspec:4]: If we have version 1.1.0 it should probably be also in here right?

**Review by olejnjak — APPROVED** [review]
Just a nitpick. 👍

**Review by IgorRosocha — APPROVED** [review]
Great job! 👍

File: OpenGoogleSignInSDK.podspec (hunk 1 above)
Comment (LukasHromadnik) [reply OpenGoogleSignInSDK.podspec:4]: nice catch! Thank you! Without it tapestry won't work. 👍

**Pull request merged**